src/main/java/software/amazon/cloudwatchlogs/emf/model/MetricsContext.java

```diff
@@ -324,25 +439,32 @@ public List<String> serialize() throws JsonProcessingException {
         } else {
             List<RootNode> nodes = new ArrayList<>();
             Map<String, Metric> metrics = new HashMap<>();
-            Queue<Metric> metricQueue = new LinkedList<>(rootNode.metrics().values());
-            while (!metricQueue.isEmpty()) {
-                Metric metric = metricQueue.poll();
+            ArrayList<Queue<Metric>> remainingMetrics = new ArrayList<>();
+            PriorityQueue<Queue<Metric>> metricQueue =
```

Comment:
Using a PriorityQueue here instead of just cycling through a queue is a slightly slower implementation but ensures that the minimum amount of metrics are flushed which the previous solution didn't. 

For example:
let MAX_METRICS_PER_EVENT = 2
let MAX_DATAPOINTS_PER_METRIC = 1

lets say we have a queue of metrics represented as how data points are in each metric
{"M1" : 1, "M2" : 1, "M3" : 2}

The previous implementation would result in 3 flushes as follows
Flushed : ["M1", "M2"]    Remaining : {"M3" : 2}
Flushed : ["M3"]              Remaining : {"M3" : 1}
Flushed : ["M3"]              Remaining : {}

The New implementation will result in 2 flushes as follows:
Flushed : ["M1", "M3"]    Remaining : {"M2" : 1, "M3" : 1}
Flushed : ["M2", "M3"]    Remaining : {}